<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genomic Dataset Indexing and Search Website Browser • XZDBfunction</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Genomic Dataset Indexing and Search Website Browser">
<meta property="og:description" content="Modularized tools for dataset indexing, gene ID/symbol resolution, searching, printing, and metadata binding. Includes full roxygen2, pkgdown, and GitHub Actions support.">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">XZDBfunction</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">7.38.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/overview.html">XZDBfunction Workflow Overview</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ScottCalcalas/XZDBfunction/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="xzdbfunction-r-package-genomic-database-website-browser">‘XZDBfunction’ R Package: Genomic Database Website Browser<a class="anchor" aria-label="anchor" href="#xzdbfunction-r-package-genomic-database-website-browser"></a>
</h1></div>
<ul>
<li>A user-friendly database and web browser application for <a href="#browser-functions">searching, annotating, and exporting</a> genomic and proteomic metadata across multiple datasets.</li>
<li>Supports gene expressing &amp; protein interaction datasets.</li>
<li>The only dependency is R ; works on Windows, macOS, and Linux. Easy to launch <a href="#instructions-to-configure-local-quick-start">instructions</a> available.</li>
<li>Also contains analysis functions for statistical and biological analysis.</li>
</ul>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Overview</th>
<th>Start Page</th>
</tr></thead>
<tbody><tr class="odd">
<td><img src="reference/figures/xzdbGraph_Explain.png" alt="Shiny UI Overview"></td>
<td><img src="reference/figures/FirstPage.png" alt="Shiny UI FirstPage"></td>
</tr></tbody>
</table>
<p>This document provides complete instructions for starting the website browser application and its functions.</p>
<div class="section level2">
<h2 id="quick-tour">Quick Tour<a class="anchor" aria-label="anchor" href="#quick-tour"></a>
</h2>
<p>(<em>Contains setup code for R, launching the web browser directly with an example dataset</em>)</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span> <span class="co"># Skip this step if you already have "remotes" package</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"scottcalcalas/XZDBfunction"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ScottCalcalas/XZDBfunction" class="external-link">XZDBfunction</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="reference/XZDB.Run.html">XZDB.Run</a></span><span class="op">(</span><span class="op">)</span> <span class="co">#Launch a demo - with example datasets</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="setup-database">Setup Database<a class="anchor" aria-label="anchor" href="#setup-database"></a>
</h2>
<p>(<em>Step-by-step setup instructions for R, and how to launch the database browser directly with example/your dataset</em>)</p>
<div class="section level3">
<h3 id="id_1-get-protocol-and-set-up-your-own-database">1. Get protocol and set up your own database<a class="anchor" aria-label="anchor" href="#id_1-get-protocol-and-set-up-your-own-database"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="fu"><a href="reference/xzdb.help.html">xzdb.help</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/xzdb.help.html">xzdb.help</a></span><span class="op">(</span><span class="op">)</span> <span class="co">#Get protocols and setup instructions</span></span>
<span></span>
<span><span class="co"># Transfer your datasets to the current folder, check everything is OK to run next step.</span></span>
<span><span class="fu"><a href="reference/xzdb.input.all.html">xzdb.input.all</a></span><span class="op">(</span><span class="op">)</span>    <span class="co">#Build the database in current folder and copy them to R package.</span></span>
<span><span class="co"># After this step, R always uses your own dataset instead of the example dataset</span></span></code></pre></div>
<p>(<a href="#setup-a-new-database">See details</a>)</p>
</div>
<div class="section level3">
<h3 id="id_2-launch-the-genomic-website-browser-in-two-different-ways">2. Launch the genomic website browser in two different ways<a class="anchor" aria-label="anchor" href="#id_2-launch-the-genomic-website-browser-in-two-different-ways"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/XZDB.Run.html">XZDB.Run</a></span><span class="op">(</span><span class="op">)</span>                 <span class="co"># launches browser; if you already ran xzdb.input.all(), it will use your own dataset instead of the example dataset</span></span>
<span></span>
<span><span class="fu"><a href="reference/XZDB.Run.html">XZDB.Run</a></span><span class="op">(</span>use_current <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>  <span class="co"># Force running on current path. Requires setup datasets.</span></span></code></pre></div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="browser-functions">Browser Functions<a class="anchor" aria-label="anchor" href="#browser-functions"></a>
</h2>
<ul>
<li><a href="#function-1-Search">1. Search</a></li>
<li><a href="#function-2-check">2. Check</a></li>
<li><a href="#function-3-data-information">3. Load Data information</a></li>
<li><a href="#function-4-output-files">4. Output Files</a></li>
<li><a href="#function-5-update-database">5. Update Database</a></li>
<li><a href="#function-6-uniprotmint-comparison">6. UniProt / MINT Comparison</a></li>
</ul>
<div class="section level3">
<h3 id="function-1-search">Function 1. Search<a class="anchor" aria-label="anchor" href="#function-1-search"></a>
</h3>
<p>Use the <strong>Search</strong> tab to obtain detailed search results.</p>
<p><br></p>
<p>Supported search options:</p>
<ul>
<li>Gene Symbol: <em>exact</em>, <em>family</em>, <em>fuzzy</em>
</li>
<li>Protein ID: <em>exact match</em>
</li>
<li>ENSEMBL Gene ID: <em>exact match</em>
</li>
</ul>
<p><em>(Family search: seaching for gene/proteins start with the same letters but haveing different numbers at end. NCBP2 family search will also includes NCBP1.)</em></p>
<p><em>(Fuzzy search: use threshold of 0.1 to have a very permissive, broad search.)</em></p>
<p><br></p>
<p>Example: exact searching <strong>CDK13</strong> and <strong>NCBP2</strong> returns all matching rows for either CDK13 or NCBP2 across all datasets.</p>
<p>The app generates three files automatically:</p>
<ul>
<li>Full results<br>
</li>
<li>Details for Gene 1<br>
</li>
<li>Details for Gene 2</li>
</ul>
<p><br></p>
<div class="float">
<img src="reference/figures/function1.png" alt="Shiny UI function1"><div class="figcaption">Shiny UI function1</div>
</div>
<p>Each line represents a row found in our current database, with that database’s description at the beginning.</p>
<p><br><br></p>
</div>
<div class="section level3">
<h3 id="function-2-check">Function 2. Check<a class="anchor" aria-label="anchor" href="#function-2-check"></a>
</h3>
<p>Use the <strong>Check</strong> tab to: - See which datasets include a specific gene<br>
- Confirm if the gene exists in the database</p>
<div class="float">
<img src="reference/figures/function2.png" alt="Shiny UI function2"><div class="figcaption">Shiny UI function2</div>
</div>
<p><br><br></p>
</div>
<div class="section level3">
<h3 id="function-3-data-information">Function 3. Data information<a class="anchor" aria-label="anchor" href="#function-3-data-information"></a>
</h3>
<p>On the <strong>Data info</strong> tab, use <strong>“Load Data info”</strong> button to view dataset metadata and structure.</p>
<div class="float">
<img src="reference/figures/function3.png" alt="Shiny UI function3"><div class="figcaption">Shiny UI function3</div>
</div>
<p><br><br></p>
</div>
<div class="section level3">
<h3 id="function-4-output-files">Function 4. Output Files<a class="anchor" aria-label="anchor" href="#function-4-output-files"></a>
</h3>
<p>Open the <strong>Output files</strong> tab to: - Preview a result file (using the red <em>Preview Selected</em> button)<br>
- Download results (using the <em>Download</em> button in each row)</p>
<p><strong>Note:</strong><br>
Not all outputs can be previewed.<br>
The output folder should be cleared monthly to prevent slowdowns—please save results you want to keep in other folders.</p>
<div class="float">
<img src="reference/figures/function4.png" alt="Shiny UI function4"><div class="figcaption">Shiny UI function4</div>
</div>
<p><br><br></p>
</div>
<div class="section level3">
<h3 id="function-5-update-database">Function 5. Update Database<a class="anchor" aria-label="anchor" href="#function-5-update-database"></a>
</h3>
<p><em>(Optional — only needed if updating dataset metadata)</em></p>
<p>After updating: - <code>datasets/</code> - <code>Datasets information.xlsx</code></p>
<p>Run “Rebuild EVERYTHING” at Administrator Operations in the left panel:</p>
<div class="float">
<img src="reference/figures/function5.png" alt="Shiny UI function5"><div class="figcaption">Shiny UI function5</div>
</div>
<p><br><br></p>
</div>
<div class="section level3">
<h3 id="function-6-uniprotmint-comparison">Function 6. UniProt/MINT Comparison<a class="anchor" aria-label="anchor" href="#function-6-uniprotmint-comparison"></a>
</h3>
<ul>
<li>
<strong>UniProt:</strong> retrieves protein entry information<br>
</li>
<li>
<strong>MINT:</strong> retrieves protein–protein interaction data based on UniProt ID</li>
</ul>
<p>Requires an internet connection.</p>
<p><strong>If you search from current result:</strong> <img src="reference/figures/function61.png" alt="Shiny UI function6.1"></p>
<p><br></p>
<p><strong>If you type the ID:</strong> <img src="reference/figures/function62.png" alt="Shiny UI function6.2"></p>
<p><br></p>
<p><strong>The summary table and the Venn diagram will be like:</strong> <img src="reference/figures/function63.png" alt="Shiny UI function6.3"></p>
<p><br><br></p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="setup-a-new-database">Setup a New Database<a class="anchor" aria-label="anchor" href="#setup-a-new-database"></a>
</h2>
<p><em>(For package user to setup database inside R)</em></p>
<p>To use helper to create your own dataset, run:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/xzdb.help.html">xzdb.help</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/xzdb.input.all.html">xzdb.input.all</a></span><span class="op">(</span><span class="op">)</span>        <span class="co"># Run this after modifying the copied files</span></span></code></pre></div>
<p>Synchronize all datasets and index files(include Datasets information.xlsx) into the package shinyapp directly</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/xzdb.sync.to.shinyapp.html">xzdb.sync.to.shinyapp</a></span><span class="op">(</span><span class="op">)</span> <span class="co">#No need to run if you already run xzdb.input.all()</span></span></code></pre></div>
<p>Get the currently used datasets inside the package location (If you want to confirm it’s copied successfully)</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/xzdb.nowDataset.html">xzdb.nowDataset</a></span><span class="op">(</span><span class="op">)</span>          <span class="co">#It copies current using datasets to your working path</span></span></code></pre></div>
<p>Example package storage location, inside R package:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a> R\R<span class="dv">-4</span>.<span class="fl">4.3</span>\library\XZDBfunction\shinyapp</span></code></pre></div>
<p><em>After those steps, start the browser and run “Rebuild EVERYTHING” at Administrator Operations.</em></p>
<hr>
</div>
<div class="section level2">
<h2 id="instructions-to-configure-local-quick-start">Instructions to Configure Local Quick Start<a class="anchor" aria-label="anchor" href="#instructions-to-configure-local-quick-start"></a>
</h2>
<p>(If you want to use it as one click, instead of opening R)</p>
<div class="section level3">
<h3 id="steps">
<em>Steps:</em><a class="anchor" aria-label="anchor" href="#steps"></a>
</h3>
<ul>
<li><a href="#id_0-configure-r-file">0. Configure R file</a></li>
<li>
<a href="#id_1-auto-start">1. Auto Start</a>
<ul>
<li><a href="#on-windows">On Windows</a></li>
<li><a href="#on-mac">On Mac</a></li>
</ul>
</li>
<li><a href="#id_2start--package-installation">2.Start &amp; Package Installation</a></li>
<li><a href="#id_3last-step">3.Last Step</a></li>
<li><a href="#appendix">Appendix — Windows &amp; Mac Shortcuts</a></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="id_0-configure-r-file">0. Configure R file<a class="anchor" aria-label="anchor" href="#id_0-configure-r-file"></a>
</h2>
<p>To make a quick start R file, put these two lines:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ScottCalcalas/XZDBfunction" class="external-link">XZDBfunction</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/XZDB.Run.html">XZDB.Run</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>For example, it can be called <strong>Quick_Start.R</strong>. The next step is just setting up to run this file using R (not RStudio).</p>
</div>
<div class="section level2">
<h2 id="id_1-auto-start">1. Auto Start<a class="anchor" aria-label="anchor" href="#id_1-auto-start"></a>
</h2>
<p><em>(Check examples at Appendix)</em></p>
<p><em>On Windows</em>:</p>
<p>A desktop shortcut can automatically launch the Genomic DB Browser, as long as <strong>R is installed</strong>.</p>
<p><em>On Mac</em>:</p>
<ol style="list-style-type: decimal">
<li>Ensure <strong>R</strong> is installed.<br>
</li>
<li>Open the project folder.<br>
</li>
<li>Double-click <strong>MAC_Start.command</strong>. <em>If the Mac start script fails, use the backup version inside the <code>App_Info</code> folder.</em>
</li>
</ol>
</div>
<div class="section level2">
<h2 id="id_2start--package-installation">2.Start &amp; Package Installation<a class="anchor" aria-label="anchor" href="#id_2start--package-installation"></a>
</h2>
<p>On the first run, the script will automatically install any missing R packages.</p>
<p>Once installation finishes, the app will open in your browser automatically.</p>
</div>
<div class="section level2">
<h2 id="id_3last-step">3.Last Step<a class="anchor" aria-label="anchor" href="#id_3last-step"></a>
</h2>
<p>After finishing: - Close the browser tab<br>
- Close R / RStudio / Terminal</p>
</div>
<div class="section level2">
<h2 id="appendix">
<em>Appendix</em><a class="anchor" aria-label="anchor" href="#appendix"></a>
</h2>
<div class="section level4">
<h4 id="a1windows-shortcut-example">A1.Windows Shortcut Example<a class="anchor" aria-label="anchor" href="#a1windows-shortcut-example"></a>
</h4>
<p>Modify the shortcut <strong>Target</strong> to point to Rscript and your Quick Start script: (For example:)</p>
<pre><code>C:\Users\jcc1885\AppData\Local\Programs\R\R-4.4.3\bin\Rscript.exe "R:\Basic_Sciences\Pharm\Borden_Lab\borden\Database\Genomic result Browser\Quick Start.R"</code></pre>
</div>
<div class="section level4">
<h4 id="a2mac-one-click-start-command-file">A2.Mac “One-Click Start” Command File<a class="anchor" aria-label="anchor" href="#a2mac-one-click-start-command-file"></a>
</h4>
<p>Create a file named <code>something.command</code> (e.g. <code>MAC_Start.command</code>) on the Desktop:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co"># Launch script</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="co"># ...</span></span></code></pre></div>
<p>Then run</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="bu">cd</span> ~/Desktop</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">chmod</span> +x something.command</span></code></pre></div>
<p>After those steps, double-click the file to launch the website browser application.</p>
</div>
<div class="section level4">
<h4 id="a3-update-local-r-script-if-youre-using-local-based-instead-of-package-based">A3. Update local R script (If you’re using local based instead of package based)<a class="anchor" aria-label="anchor" href="#a3-update-local-r-script-if-youre-using-local-based-instead-of-package-based"></a>
</h4>
<p>Copy and replace your local scripts using the scripts at github location: XZDBfunction/inst/shinyapp/</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/ScottCalcalas/XZDBfunction/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ScottCalcalas/XZDBfunction/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing XZDBfunction</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Xiaopei Zhang <br><small class="roles"> Maintainer </small>   </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Xiaopei Zhang.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
